src/inc/quic_datapath.h

```diff
@@ -555,7 +555,8 @@ CXPLAT_SEND_DATA*
 CxPlatSendDataAlloc(
     _In_ CXPLAT_SOCKET* Socket,
     _In_ CXPLAT_ECN_TYPE ECN,
-    _In_ uint16_t MaxPacketSize
+    _In_ uint16_t MaxPacketSize,
+    _In_ QUIC_ADDRESS_FAMILY Family
```

Comment:
I'd prefer if we could have a design that doesn't require this (though I'm not 100% against the change). I haven't looked in detail at the change yet, but I assume this is done to allocate exactly enough backfill, based on the selected address family. What if, instead, we simply always allocated enough space for the largest we'd ever need? Or do we think this would waste too much space?

Reply:
We would 100% need to do perf testing on this. We've attempted changes that add extra args on the send path, and it has sometimes cost 10% of performance. I feel like a few extra bytes isn't a big deal when the alternative is this invasive of a change not normally used in any other datapath.

Reply:
Right, we need to know how much space the header needs so that the offset for the payload can be determined. Since XDP packets are all just pre-allocated in the size of MAX_ETH_FRAME_SIZE, if we always allocate say 62 (IPv6+Eth+UDP), then we will waste 20 bytes because payload is written before headers. So, I think my current change looks very natural to me.

I am surprised that an extra argument could cause 10% perf loss. In any other datapath, this parameter should be optimized out.

Reply:
I'm also surprised to hear that just adding args in a datapath function caused a 10% perf regression. Are you sure the regression wasn't actually from whatever logic was related to the added args?